_overviews/toolkit/web-server-input.md

```diff
@@ -0,0 +1,232 @@
+        |  <label for="name">First name:</label><br>
+        |  <input type="text" name="name" value=""><br>
+        |  <label for="surname">Last name:</label><br>
+        |  <input type="text" name="surname" value=""><br><br>
+        |  <input type="submit" value="Submit">
+        |</form>
+        |</body>
+        |</html>""".stripMargin
+
+    cask.Response(data = html, headers = Seq("Content-Type" -> "text/html"))
+
+  @cask.postForm("/form")
+  def formEndpoint(name: String, surname: String) =
+    "Hello " + name + " " + surname
+
+  initialize()
+```
+{% endtab %}
+{% endtabs %}
+
+## Handling JSON-encoded input
+
+JSON fields are handled in the same way as form fields, except that `cask.PostJson` annotation is used. The topmost fields
+will be read into the endpoint method arguments and if any of them is missing or has an incorrect type, an error message
+will be returned with 400 response code.
+
+{% tabs web-server-input-2 class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+object MyApp extends cask.MainRoutes {
+  
+  @cask.postJson("/json")
+  def jsonEndpoint(name: String, surname: String) =
+    "Hello " + name + " " + surname
+
+  initialize()
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+object MyApp extends cask.MainRoutes:
+
+  @cask.postJson("/json")
+  def jsonEndpoint(name: String, surname: String) = 
+    "Hello " + name + " " + surname
+  
+  initialize()
+```
+{% endtab %}
+{% endtabs %}
+
+Send the POST request using `curl`:
+
+```shell
+curl --header "Content-Type: application/json" \
+  --data '{"name":"John","surname":"Smith"}' \
+  http://localhost:8080/json
+```
+
+The response will be:
+```
+Hello John Smith
+```
+
+Deserialization is handled by uPickle JSON library. To deserialize an object, use `ujson.Value` type.
+
+{% tabs web-server-input-3 class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+object MyApp extends cask.MainRoutes {
+
+  @cask.postJson("/json")
+  def jsonEndpoint(value: ujson.Value) =
+    value.toString
+
+  initialize()
+}
+
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+object MyApp extends cask.MainRoutes:
+
+  @cask.postJson("/json")
+  def jsonEndpoint(value: ujson.Value) = 
+    value.toString
+
+  initialize()
+
+```
+{% endtab %}
+{% endtabs %}
+
+Send a POST request.
+```shell
+curl --header "Content-Type: application/json" \
+  --data '{"value":{"name":"John","surname":"Smith"}}' \
+  http://localhost:8080/json2
+```
+
+Server will respond with:
+```
+"{\"name\":\"John\",\"surname\":\"Smith\"}"
+```
+
+## Handling JSON-encoded output
+
+Cask endpoint can return JSON objects returned by uPickle library functions. Cask will automatically handle the `ujson.Value`
+type and set the `Content-Type application/json` header.
+
+In this example we use a simple `TimeData` case class to send information about the time zone and current time in a chosen
+location. To serialize a case class into JSON you need to define a serializer in its companion object.
```

Comment:
There's "we" in the previous sentence and "you" in this one. We should be consistent throughout.